runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/DistributionCell.java

```diff
@@ -80,6 +84,7 @@ public DirtyState getDirty() {
 
   @Override
   public DistributionData getCumulative() {
+    value.get().computePercentiles();
```

Comment:
I think we can just return as value.get(). the actual percentile metrics should be generated by distributiondata.percentile() call, right? If that is the only place to get percentile, we don't need to compute it here.

Reply:
Changed.